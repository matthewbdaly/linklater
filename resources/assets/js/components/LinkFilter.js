// @flow
import React, { Component } from 'react';
import ReactDOM from 'react-dom';

type Props = {
  updateFilter: (filter: string) => void
};

export default class LinkFilter extends Component<Props> {
  filterRef: ?React.createRef<'input'>;
  clearRef: ?React.createRef<'button'>;
  updateFilter: () => void;
  clearFilter: () => void;
  constructor(props: Props) {
    super(props);
    this.filterRef = React.createRef();
    this.updateFilter = this.updateFilter.bind(this);
    this.clearRef = React.createRef();
    this.clearFilter = this.clearFilter.bind(this);
  }
  updateFilter(e: SyntheticEvent<HTMLInputElement>) {
    e.preventDefault();
    let filter = this.filterRef.current.value.trim();
    this.props.updateFilter(filter);
  }
  clearFilter(e: SyntheticEvent<HTMLButtonElement>) {
    e.preventDefault();
    this.filterRef.current.value = '';
    this.props.updateFilter('');
  }
  render() {
    return (
      <div className="form-group">
        <input id="linkfilter" placeholder="Filter links..." className="form-control" ref={this.filterRef} onChange={this.updateFilter} />
        <button id="clear" className="btn btn-primary" ref={this.clearRef} onClick={this.clearFilter}>Clear filter</button>
      </div>
    );
  }
}
