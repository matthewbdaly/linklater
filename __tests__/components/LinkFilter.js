import React from 'react';
import {findDOMNode} from 'react-dom';
import LinkFilter from '../../resources/assets/js/components/LinkFilter';
import renderer from 'react-test-renderer';

describe('LinkFilter', () => {
  it('renders the link filter', () => {
    const update = jest.fn();
    const tree = renderer
      .create(<LinkFilter updateFilter={update} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
