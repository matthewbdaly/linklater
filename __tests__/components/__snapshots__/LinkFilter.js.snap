// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkFilter renders the link filter 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    id="linkfilter"
    onChange={[Function]}
    placeholder="Filter links..."
  />
  <button
    className="btn btn-primary"
    id="clear"
    onClick={[Function]}
  >
    Clear filter
  </button>
</div>
`;
